XenAdmin/Controls/SrPickerItem.cs

```diff
@@ -131,8 +131,10 @@ protected override string CannotBeShownReason
                     return Messages.CURRENT_LOCATION;
                 if (LocalToLocalMove())
                     return Messages.LOCAL_TO_LOCAL_MOVE;
-                if (!SrIsLocalToTheHostOnForExsistingVDIs())
+                if (TheSR.IsLocalSR && !SrIsLocalToTheHostOnForExsistingVDIs())
                     return Messages.SRPICKER_ERROR_LOCAL_SR_MUST_BE_RESIDENT_HOSTS;
+                if (!TheSR.CanBeSeenFrom(Affinity))
+                    return string.Format(Messages.SR_CANNOT_BE_SEEN, Affinity == null ? Helpers.GetName(TheSR.Connection) : Helpers.GetName(Affinity));

```

Comment:
In Helpers.GetName(TheSR.Connection), can the connection here be null? If so, it will return empty string and an unclear message will be shown.